### ISSUE TITLE
Track lengths obtained by BetterRYM do not match track lengths displayed on scraped website

### ISSUE THREAD

**Issue by FlorianWolters** [jgchk/better-rym#111]
This issue occurs at least for Bandcamp, Spotify and Apple Music. In "Step 0" a URL of one of these sites is entered and "Import" is selected. By doing this BetterRYM fills out the "Track listing". But in almost all cases there are one (more often multiple) track lengths that do not match the track lengths displayed on the website.

Example: https://music.apple.com/us/album/inevitable-demise/1529485989

Apple Music Website:

    1|Sworn and Forlorn|4:58
    2|Ater Mortem|3:42
    3|Bewitched|4:13
    4|Inevitable Demise|4:36
    5|Deep in the Vessel|6:16
    6|Swarm of Pest|3:06
    7|Pandemonium|4:48
    8|Blind Obediance|4:17
    9|Defeat|7:14

BetterRYM:

    1|Sworn and Forlorn|4:58
    2|Ater Mortem|3:42
    3|Bewitched|4:14
    4|Inevitable Demise|4:37
    5|Deep in the Vessel|6:16
    6|Swarm of Pest|3:07
    7|Pandemonium|4:48
    8|Blind Obediance|4:17
    9|Defeat|7:14

BetterRYM is super helpful and helps to avoid a lot of mistakes during submission. But this issue requires that the BetterRYM user manually re-checks each and every track length for a submission.

Is there any "magic" implemented that derives the track lengths from another source? If so, I think this should be disabled. Imo, BetterRYM should always return the track lengths that are displayed on the website, otherwise its hard/impossible for RYM moderators to verify the track lengths.

**Comment by jgchk** [comment]
howdy! betterrym uses the millisecond track lengths from these sources and rounds them to the nearest full second. the discrepancy is due to these services always rounding down. i believe rounded track lengths are preferred for db entries, though i'd be happy to change this behavior if there are guidelines that say otherwise

**Comment by FlorianWolters** [comment]
Alright. I've expected something as this. I don't know of any RYM guidelines.

> i believe rounded track lengths are preferred for db entries

Can you provide a link for that assumption? I would assume "What You See Is What You Get" is good common sense, also for track lengths. But I may be wrong.

My reasoning is: Take the track lengths printed on a physical release for example. I am not aware of any physical release that has track lengths printed on it with a time resolutions < 1s. No one would rip/digitize the physical release to obtain the maximum time resolution (even that couldn't be accurate enough for analog recordings because of manual editing).

I think the difference between what is shown on a website (physical release) and submitted on RYM only introduces unnecessary discussions and/or following unnecessary correction submissions that try to "fix" track lengths.

I consider simply using the track lengths shown to the user as the "right thing".

But as I said: I may be completely wrong if there is any rule/guideline on RYM. Please provide me with a link if you are aware of such a thing.

**Comment by jgchk** [comment]
yeah this is a bit of a gray area in terms of guidelines. i don't know of any written rules on the subject either. i've implemented it this way in order to be as accurate as possible, and to make rounding consistent between streaming platforms. you make a fair point that the discrepancy with what is displayed on streaming services can introduce some confusion.

for now i'll be keeping rounded track lengths, but I'll use this github issue to add a toggleable option for rounded vs floored track lengths in the (hopefully near) future
